forui/lib/src/widgets/line_calendar/line_calendar.dart

```diff
@@ -0,0 +1,346 @@
+        selectedDecoration = BoxDecoration(
+          color: colorScheme.primary,
+          borderRadius: style.borderRadius,
+        ),
+        unselectedDecoration = BoxDecoration(
+          color: colorScheme.background,
+          borderRadius: style.borderRadius,
+          border: Border.all(color: colorScheme.border),
+        ),
+        content = FLineCalendarContentStyle.inherit(colorScheme: colorScheme, typography: typography);
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('itemPadding', itemPadding))
+      ..add(DiagnosticsProperty('selectedCurrentDateIndicatorColor', selectedCurrentDateIndicatorColor))
+      ..add(DiagnosticsProperty('unselectedCurrentDateIndicatorColor', unselectedCurrentDateIndicatorColor))
+      ..add(DiagnosticsProperty('selectedDecoration', selectedDecoration))
+      ..add(DiagnosticsProperty('unselectedDecoration', unselectedDecoration))
+      ..add(DiagnosticsProperty('content', content));
+  }
+
+  /// Returns a copy of this [FLineCalendarStyle] with the given properties replaced.
+  @useResult
+  FLineCalendarStyle copyWith({
+    double? itemPadding,
+    Color? selectedCurrentDateIndicatorColor,
+    Color? unselectedCurrentDateIndicatorColor,
+    BoxDecoration? selectedDecoration,
+    BoxDecoration? unselectedDecoration,
+    FLineCalendarContentStyle? content,
+  }) =>
+      FLineCalendarStyle(
+        itemPadding: itemPadding ?? this.itemPadding,
+        selectedCurrentDateIndicatorColor: selectedCurrentDateIndicatorColor ?? this.selectedCurrentDateIndicatorColor,
+        unselectedCurrentDateIndicatorColor:
+            unselectedCurrentDateIndicatorColor ?? this.unselectedCurrentDateIndicatorColor,
+        selectedDecoration: selectedDecoration ?? this.selectedDecoration,
+        unselectedDecoration: unselectedDecoration ?? this.unselectedDecoration,
+        content: content ?? this.content,
+      );
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is FLineCalendarStyle &&
+          runtimeType == other.runtimeType &&
+          itemPadding == other.itemPadding &&
+          selectedCurrentDateIndicatorColor == other.selectedCurrentDateIndicatorColor &&
+          unselectedCurrentDateIndicatorColor == other.unselectedCurrentDateIndicatorColor &&
+          selectedDecoration == other.selectedDecoration &&
+          unselectedDecoration == other.unselectedDecoration &&
+          content == other.content;
+
+  @override
+  int get hashCode =>
+      itemPadding.hashCode ^
+      selectedCurrentDateIndicatorColor.hashCode ^
+      unselectedCurrentDateIndicatorColor.hashCode ^
+      selectedDecoration.hashCode ^
+      unselectedDecoration.hashCode ^
+      content.hashCode;
+}
+
+/// A line calendar's content style.
+final class FLineCalendarContentStyle with Diagnosticable {
```

Comment:
nitpick: this style should be moved into line_calendar_tile.

Also it has the same issues as `FLineCalendarStyle`.